[macOS] WPT test `html/rendering/replaced-elements/the-select-element/select-1-line-height.html` is failing

lilyspiniolas opened the pull request:
#### b83fa6f790936a153f81eacb217e246a684c5da1
<pre>
[macOS] WPT test `html/rendering/replaced-elements/the-select-element/select-1-line-height.html` is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=301611">https://bugs.webkit.org/show_bug.cgi?id=301611</a>
<a href="https://rdar.apple.com/162394818">rdar://162394818</a>

Reviewed by Aditya Keerthi and Richard Robinson.

Enforce `line-height: normal` during style adjustment for native select
controls on macOS. This behavior matches iOS and is similar to what we
did prior to the form control refresh on macOS where we forced line-height
to `RenderStyle::initialLineHeight()`. This preserves a historic behavior
seen across browsers where `line-height` has no impact on select elements
which render as drop-down boxes.

Renamed `RenderThemeCocoa::paintInnerSpinButtonStyleForVectorBasedControls()`
to `RenderThemeCocoa::paintInnerSpinButtonForVectorBasedControls()`.

Tested by `imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-1-line-height.html`

* LayoutTests/platform/mac-sequoia/fast/forms/control-restrict-line-height-expected.txt: Renamed from LayoutTests/platform/mac-sequoia-wk1/fast/forms/control-restrict-line-height-expected.txt.
* LayoutTests/platform/mac-sonoma/fast/forms/control-restrict-line-height-expected.txt: Renamed from LayoutTests/platform/mac-sonoma-wk1/fast/forms/control-restrict-line-height-expected.txt.
* LayoutTests/platform/mac-wk1/fast/forms/control-restrict-line-height-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/fast/forms/control-restrict-line-height-expected.txt: Removed.
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/fast/forms/control-restrict-line-height-expected.txt:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintInnerSpinButtonForVectorBasedControls):
(WebCore::adjustSelectListButtonStyleForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustMenuListStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustMenuListButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::paintInnerSpinButton):
(WebCore::RenderThemeCocoa::paintInnerSpinButtonStyleForVectorBasedControls): Deleted.
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::adjustSelectListButtonStyle):

Canonical link: <a href="https://commits.webkit.org/302428@main">https://commits.webkit.org/302428@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/74e6a6c7fbb5e9c0e873b4e85662e32e36939c0f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129077 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1336 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39913 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136457 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/80451 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4a769137-e816-40bc-8256-c11307e5c9e2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/130948 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1267 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1213 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98276 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/80451 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4dbe31a9-b038-4c26-a1e9-75e09c9cf640) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132024 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/976 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115624 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78922 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/aa33daed-3743-48f9-b255-3277bc3771d0) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/900 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33739 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79736 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109345 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34237 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138930 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1129 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1097 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106814 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1181 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111960 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106641 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27146 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/924 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30484 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53643 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1202 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64540 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1028 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1074 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1126 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
```diff
@@ -2649,21 +2633,24 @@ static void adjustInputElementButtonStyleForVectorBasedControls(RenderStyle& sty
     }
 
+    // Enforce "line-height: normal".
+    style.setLineHeight(CSS::Keyword::Normal { });
```
I think this will affect date inputs (on iOS) too, is that intentional?

aprotyas replied:
this also affects other menu list buttons right? like `<input type="date/time/month/week">`?

aprotyas reviewed (commented):
It seems like the test is failing in stress mode

```
FAIL: got computed line-height '100px', expected 'normal' or a length <= 50px
```

pxlcoder commented on Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
```diff
@@ -2505,6 +2495,18 @@ static void adjustInputElementButtonStyleForVectorBasedControls(RenderStyle& sty
+#else
+    UNUSED_PARAM(element);
+#endif
+    // Enforce "line-height: normal".
```
This comment describes the code. It can be removed.

pxlcoder commented on Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
```diff
@@ -2521,6 +2523,10 @@ static void adjustInputElementButtonStyleForVectorBasedControls(RenderStyle& sty
     style.setTextWrapMode(TextWrapMode::NoWrap);
     style.setBoxShadow(CSS::Keyword::None  { });
 
+    // Enforce "line-height: normal" as long as this element isn't a non-select element using `-webkit-appearance: menulist`.
```
Ditto.

pxlcoder commented on Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
```diff
@@ -2649,6 +2655,13 @@ static void adjustInputElementButtonStyleForVectorBasedControls(RenderStyle& sty
 
+    // Enforce some default styles in the case that this is a non-multiple <select> element
+    // on any Cocoa platform, or a date input on iOS. We don't force these if this is just
+    // an element with "-webkit-appearance: menulist-button".
```
I think this comment can be removed too. 

> We don't force these if this is just an element with "-webkit-appearance: menulist-button"

Is particularly unnecessary now that we no longer allow arbitrary HTML elements to have `appearance`.

lilyspiniolas committed:
[macOS] WPT test `html/rendering/replaced-elements/the-select-element/select-1-line-height.html` is failing
https://bugs.webkit.org/show_bug.cgi?id=301611
rdar://162394818

Reviewed by Aditya Keerthi and Richard Robinson.

Enforce `line-height: normal` during style adjustment for native select
controls on macOS. This behavior matches iOS and is similar to what we
did prior to the form control refresh on macOS where we forced line-height
to `RenderStyle::initialLineHeight()`. This preserves a historic behavior
seen across browsers where `line-height` has no impact on select elements
which render as drop-down boxes.

Renamed `RenderThemeCocoa::paintInnerSpinButtonStyleForVectorBasedControls()`
to `RenderThemeCocoa::paintInnerSpinButtonForVectorBasedControls()`.

Tested by `imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-1-line-height.html`

* LayoutTests/platform/mac-sequoia/fast/forms/control-restrict-line-height-expected.txt: Renamed from LayoutTests/platform/mac-sequoia-wk1/fast/forms/control-restrict-line-height-expected.txt.
* LayoutTests/platform/mac-sonoma/fast/forms/control-restrict-line-height-expected.txt: Renamed from LayoutTests/platform/mac-sonoma-wk1/fast/forms/control-restrict-line-height-expected.txt.
* LayoutTests/platform/mac-wk1/fast/forms/control-restrict-line-height-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/fast/forms/control-restrict-line-height-expected.txt: Removed.
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/fast/forms/control-restrict-line-height-expected.txt:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintInnerSpinButtonForVectorBasedControls):
(WebCore::adjustSelectListButtonStyleForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustMenuListStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustMenuListButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::paintInnerSpinButton):
(WebCore::RenderThemeCocoa::paintInnerSpinButtonStyleForVectorBasedControls): Deleted.
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::adjustSelectListButtonStyle):

Canonical link: https://commits.webkit.org/302428@main
